### PR TITLE
Additional documentation for FieldsCombinationValidator and  FieldValidator

### PR DESCRIPTION
**Pull request by RomanShostakSkai**
After some time of using PL validators we still find it confusing what every method of PL validator is about.
This is an attempt to improve javadoc on them.

@kenshoo/ecom-walmart

### CODE REVIEW

**Review by HannaSeraya — APPROVED** [review]

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
 public interface FieldsCombinationValidator<E extends EntityType<E>> extends ChangeValidator {
 
     /**
-     * @return the list of fields whose combination is being validated
+     * @return the list of fields whose combination is being validated; Can contains only current entity fields.
```
Comment (ilya0508) [inline main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:21]: **contain**

Comment (ilya0508) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:21]: What happens if developer puts to the stream fields from other entities? I think we need to explain it here

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java [hunk 2]
```diff
@@ -33,14 +33,14 @@
     ValidationError validate(FieldsValueMap<E> fieldsValueMap);
 
     /**
-     * @return a list of fields to fetch.
+     * @return a list of fields to fetch. Can contain only parent entities fields.
```
Comment (ilya0508) [inline main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:36]: If this is true, I suggest to rename the method as it is very confusing. @dimagorelik WDYT? I know this is a big refactoring, but... 

Also, what happens if there fields from current entity placed to this stream by developer?

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java [hunk 3]
```diff
@@ -33,14 +33,14 @@
...
      */
     default Stream<EntityField<?, ?>> fetchFields() {
         return Stream.of();
     }
 
     /**
-     * @return Predicate when should validate fields.
+     * @return Predicate when should validate fields. It works with fetchFields(), so Only parent fields can be used here.
```
Comment (ilya0508) [inline main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:43]: can we rephrase it to
`It works in pair with fetchFields(), only fetchFields() fields can be used by predicate.`

What happens if predicate uses some field not present in fetchFields()?

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java (hunk 1 above)
Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:21]: @RomanShostakSkai 
The developer couldn't do it here because the field belongs to generic E

**Review by dimagorelik — APPROVED** [review]

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java (hunk 3 above)
Comment (effiban) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:43]: I would phrase it like this:
 "It is used together with `fetchFields()`, so only parent fields can be referenced here."

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java (hunk 1 above)
Comment (effiban) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java:21]: > **contain**

:+1: , I would rephrase it like this:
`May contain fields of the current entity only.`

Comment (RomanShostakSkai) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java]: @effiban I have changed as you proposed.

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java (hunk 2 above)
Comment (RomanShostakSkai) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java]: @effiban I have changed as you proposed.

File: main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java (hunk 3 above)
Comment (RomanShostakSkai) [reply main/src/main/java/com/kenshoo/pl/entity/spi/FieldsCombinationValidator.java]: @effiban I have changed as you proposed.
